### PR TITLE
[docs] Document storage limitations with CSI proxy

### PR DESCRIPTION
**Pull request by sebsoto**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -233,7 +233,9 @@ config for the OpenShift Container Platform. WMCO will not be able to automatica
 workloads.
 
 ### Storage
-At this time, only in-tree storage is supported in all cloud providers.
+Windows Nodes are running csi-proxy and are ready to use CSI drivers, however Windows CSI driver DaemonSets are not
+managed by the cluster. In order to use persistent storage for Windows workloads, the cluster administrator should
```
Comment (jrvaldes) [inline README.md:237]:
```suggestion
Windows Nodes are running csi-proxy and are ready to use CSI drivers. However, the Windows CSI driver DaemonSets is not
managed by the Windows Machine Config Operator or cluster component. To use persistent storage for Windows workloads, the cluster administrator should
```

**Review by jrvaldes — COMMENTED** [review]
@sebsoto thanks for working on this, PTAL at the suggestions.

**Review by jrvaldes — APPROVED** [review]
/lgtm

**Comment by mansikulkarni96** [comment]
LGTM

File: README.md [hunk 2]
```diff
@@ -233,7 +233,9 @@ config for the OpenShift Container Platform. WMCO will not be able to automatica
...
+follow upstream documentation to deploy the Windows CSI driver DaemonSet for the desired storage provider.
```
Comment (aravindhp) [inline README.md:238]:
```suggestion
follow upstream or cloud provider documentation to deploy the Windows CSI driver DaemonSet for the desired storage provider.
```

File: README.md (hunk 1 above)
Comment (aravindhp) [inline README.md:237]: I am not sure about the `Windows CSI driver DaemonSets are not managed by the cluster`. Maybe `Windows CSI drivers are not deployed as part of the product` or something along those lines.

**Review by aravindhp — CHANGES_REQUESTED** [review]
We also need to clarify our support stance with the CSI driver but maybe that is not required here. @DCChadwick can clearly call that out in the OpenShift docs.

File: README.md (hunk 2 above)
Comment (sebsoto) [reply README.md]: Reworded, but not all storage drivers == cloud provider. For example SMB

**Review by aravindhp — COMMENTED** [review]
Where are we going to document the upgrade workflow?

**Review by saifshaikh48 — COMMENTED** [review]
/lgtm

**Comment by sebsoto** [comment]
> Where are we going to document the upgrade workflow?

Opened a separate PR for that, https://github.com/openshift/windows-machine-config-operator/pull/1964
This will be different for release-4.14 and release-4.13 because vsphere and azure switch from in tree at different OCP releases.

**Comment by sebsoto** [comment]
/override ci/prow/aws-e2e-operator ci/prow/aws-e2e-upgrade ci/prow/azure-e2e-operator ci/prow/azure-e2e-upgrade ci/prow/ci-bundle-wmco-bundle ci/prow/gcp-e2e-operator ci/prow/nutanix-e2e-operator ci/prow/nutanix-e2e-operator ci/prow/images ci/prow/vsphere-proxy-e2e-operatorci/prow/wicd-unit-vsphere

**Comment by sebsoto** [comment]
/override ci/prow/aws-e2e-operator ci/prow/aws-e2e-upgrade ci/prow/azure-e2e-operator ci/prow/azure-e2e-upgrade ci/prow/ci-bundle-wmco-bundle ci/prow/gcp-e2e-operator ci/prow/nutanix-e2e-operator ci/prow/nutanix-e2e-operator ci/prow/images ci/prow/vsphere-proxy-e2e-operator ci/prow/wicd-unit-vsphere

**Comment by sebsoto** [comment]
/override ci/prow/platform-none-vsphere-e2e-operator ci/prow/vsphere-e2e-operator

**Pull request merged**